homeassistant/helpers/config_entry_oauth2_flow.py

```diff
@@ -239,6 +240,8 @@ async def async_step_pick_implementation(
             return await self.async_step_auth()
 
         if not implementations:
+            if self.DOMAIN in await async_get_application_credentials(self.hass):
+                return self.async_abort(reason="missing_credentials")
```

Comment:
Do we need a common translation string?
https://github.com/home-assistant/core/blob/513e276bbaba4c1d750cad4253db5c79fb341ec5/homeassistant/strings.json#L73

Reply:
Typically on the abort step, we render a core driven string with a "Close" dialog.  With missing_credentials in the frontend PR this is showing its own dialog with a "Yes/No" prompt and is using a frontend translation.

https://github.com/home-assistant/frontend/pull/12708/files#diff-e67939fd25c650222db710f18764d10ae69454b0e8ad680f5e10177c2db93ceaR2806

<img width="597" alt="1 - Missing creds" src="https://user-images.githubusercontent.com/6026418/168749171-1e1ee698-1dbc-41b8-9a97-d52a303a4b20.png">

It looks like the dialog could be updated to use a core driven translation. Would it be preferred to use the core string here? I didn't necessarily do it this way intentionally, I was just copying a different dialog.

Reply:
I'll pass the question to @balloob.

Reply:
The `reason` is not directly used for translation this way, however, we have more of those cases (and still provide translations). Considering all reasons uses in core implementation have a matching generic transition; it makes sense to just add it. It will not harm and adds consistency.

Reply:
I've added a translation in oauth2_missing_credentials though it is different from the frontend translation since the context is different. It's just saying that the component requires application credentials, but doesn't include the question.

I'm assuming we don't need to add unused translation keys to all the integrations supporting application credentials, so not doing that.

Let me know if you had something else in mind.